core/src/services/s3/backend.rs

```diff
@@ -986,10 +996,13 @@ impl Accessor for S3Backend {
     }
 
     async fn write(&self, path: &str, args: OpWrite) -> Result<(RpWrite, Self::Writer)> {
-        Ok((
-            RpWrite::default(),
-            S3Writer::new(self.core.clone(), path, args),
-        ))
+        let s3_writer = match self.core.write_buffer_size {
+            Some(buffer_size) => {
+                S3Writer::new_with_buffer_size(self.core.clone(), path, args, buffer_size)
```

Comment:
We can just pass a `buffer_size: Option<usize>` to the constructor, and let the buffer_size being parsed like

```Rust
buffer_size: buffer_size.unwrap_or(8 * 1024 * 1024);
```

instead of providing a new `new_with_buffer_size`.

Reply:
LGTM